# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.cpp
# Commit: 78561892d142
# Full Hash: 78561892d142292d783d9adca93c72da7d45be87
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-10-10 21:33:39
# Description:
#   Bug 1857986 - add null check for mCanvasStyle r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190546
# ==============================================================================

diff -r 0fd304fa622b -r 78561892d142 dom/canvas/CanvasRenderingContext2D.cpp
--- a/dom/canvas/CanvasRenderingContext2D.cpp	Tue Oct 10 12:49:38 2023 +0300
+++ b/dom/canvas/CanvasRenderingContext2D.cpp	Tue Oct 10 09:26:22 2023 +0000
@@ -2848,7 +2848,7 @@
   WritingMode GetWritingModeForType(Type aType) const override {
     switch (aType) {
       case Type::This:
-        return WritingMode(mCanvasStyle);
+        return mCanvasStyle ? WritingMode(mCanvasStyle) : WritingMode();
       case Type::Root:
         return GetWritingMode(mPresContext->Document()->GetRootElement());
       default: